### ISSUE TITLE
PUT requests to the content type API are ignoring changes to fields

### ISSUE THREAD

**Issue by swicken-dotcms** [dotCMS/core#19986]
**Describe the bug**
PUT requests to the content type API are ignoring changes to fields. 

https://dotcms.zendesk.com/agent/tickets/103634

dotCMS Version: 22.01

**To Reproduce**

1. Create a content type (with a few fields) via a post to the API as per the documentation: https://dotcms.com/docs/latest/content-type-api 

`curl -v -u admin@dotcms.com:admin -XPOST http://localhost:8080/api/v1/contenttype -H "Content-Type: application/json" --data @bodyCreate.json`

**bodyCreate.json:** 
```json
{
    "clazz": "com.dotcms.contenttype.model.type.ImmutableSimpleContentType", 
    "defaultType": false, 
    "name": "Test Content Type", 
    "description": "THE DESCRIPTION", 
    "host": "48190c8c-42c4-46af-8d1a-0cd5db894797", 
    "owner": "dotcms.org.1", 
    "variable": "testContentType",
    "fixed": false, 
    "system": false, 
    "folder": "SYSTEM_FOLDER",
    "fields": [
            {
                "contentTypeId": "4c441ada-944a-43af-a653-9bb4f3f0cb2b",
                "dataType": "SYSTEM",
                "dbColumn": "system_field1",
                "fieldVariables": [],
                "fixed": false,
                "iDate": 1308941714000,
                "clazz": "com.dotcms.contenttype.model.field.ImmutableHostFolderField",
                "indexed": true,
                "inode": "ee91a37d-9d9f-4488-a1ce-50516e5309dc",
                "listed": false,
                "modDate": 1478557845000,
                "name": "Host/Folder",
                "readOnly": false,
                "required": true,
                "searchable": true,
                "sortOrder": 1,
                "unique": false,
                "variable": "hostfolder"
            },
            {
                "contentTypeId": "4c441ada-944a-43af-a653-9bb4f3f0cb2b",
                "dataType": "LONG_TEXT",
                "dbColumn": "text_area2",
                "fieldVariables": [],
                "fixed": false,
                "iDate": 1453474468000,
                "clazz": "com.dotcms.contenttype.model.field.ImmutableTagField",
                "indexed": true,
                "inode": "0030b173-41c4-40ae-8331-1c438fcb514e",
                "listed": false,
                "modDate": 1478557845000,
                "name": "Tags",
                "readOnly": false,
                "required": false,
                "searchable": true,
                "sortOrder": 3,
                "unique": false,
                "variable": "tags"
            }],
            "workflow": ["d61a59e1-a49c-46f2-a929-db2b4bfa88b2"]
}
```

2. Perform a GET request to get that content type by the ID that was returned
`curl -v -u admin@dotcms.com:admin -XGET http://localhost:8080/api/v1/contenttype/id/831b507d-06bb-497d-8c26-56cb880de61f` remove one of the fields from the returned array, modify the name of one of the other fields and remove the 'layout' array. Save this as **bodyUpdate.json**

**NOTE: This is important as you need the Field IDs in your PUT request to match those on the fields on the content type you created.**

3. Attempt to update the fields on the content type using a PUT request
`curl -v -u admin@dotcms.com:admin -XPUT http://localhost:8080/api/v1/contenttype/id/831b507d-06bb-497d-8c26-56cb880de61f -H "Content-Type: application/json" --data @bodyUpdate.json`

**bodyUpdate.json:**
```json
{
    "baseType": "CONTENT",
    "clazz": "com.dotcms.contenttype.model.type.ImmutableSimpleContentType",
    "defaultType": false,
    "description": "THE DESCRIPTION",
    "fields": [
        {
            "clazz": "com.dotcms.contenttype.model.field.ImmutableHostFolderField",
            "contentTypeId": "831b507d-06bb-497d-8c26-56cb880de61f",
            "dataType": "SYSTEM",
            "fieldType": "Host-Folder",
            "fieldTypeLabel": "Site or Folder",
            "fieldVariables": [],
            "fixed": false,
            "iDate": 1308941714000,
            "id": "9d1b4f4c-299b-4719-95d8-9e7e3b7c0e60",
            "indexed": true,
            "listed": false,
            "modDate": 1614021666000,
            "name": "Host/Folder CHANGED",
            "readOnly": false,
            "required": true,
            "searchable": true,
            "sortOrder": 1,
            "unique": false,
            "variable": "hostfolder"
        }
    ],
    "fixed": false,
    "folder": "SYSTEM_FOLDER",
    "host": "48190c8c-42c4-46af-8d1a-0cd5db894797",
    "iDate": 1614021666000,
    "id": "831b507d-06bb-497d-8c26-56cb880de61f",
    "multilingualable": false,
    "name": "Test Content Type",
    "owner": "dotcms.org.1",
    "system": false,
    "systemActionMappings": {},
    "variable": "testContentType",
    "versionable": true,
    "workflows": [
        {
            "archived": false,
            "creationDate": 1614021670408,
            "defaultScheme": false,
            "description": "",
            "entryActionId": null,
            "id": "d61a59e1-a49c-46f2-a929-db2b4bfa88b2",
            "mandatory": false,
            "modDate": 1613581987052,
            "name": "System Workflow",
            "system": true
        }
    ]
}
```

After the PUT call, do another GET and you will see the returned object still contains the removed field, and the name on the modified field will remain unchanged. 

**Expected behavior**
After a PUT call any removed fields should be removed from the content type, and any modified fields should have the modifications applied. 

**Desktop (please complete the following information):**
 - OS: Any
 - Browser: Any
 - Version: 22.01

**Workaround**
Currently you can delete/modify fields by using the fields API directly. 

https://dotcms.com/docs/latest/fields-content-type-api

**Comment by john-thomas-dotcms** [comment]
Draft/unfinished PR from @jdotcms here: https://github.com/dotCMS/core/pull/19989/files

**Comment by jdotcms** [comment]
I just confirmed that the issue still happening on master

**Comment by hmoreras** [comment]
When I try to edit a content type I gent an error: 

![image](https://user-images.githubusercontent.com/31667212/134584830-fd08d1e5-d0e0-425f-8fc4-857c1386d6a2.png)

**Comment by bryanboza** [comment]
Fixed, tested on release-21.10 // Docker // FF

**Comment by swicken-dotcms** [comment]
Not being backported due to its complexity
